### ISSUE TITLE
rocketchat: panic: interface conversion: interface {} is nil, not map[string]interface {}

### ISSUE THREAD

**Issue by KFC0803** [42wim/matterbridge#992]
**Describe the bug**
crashes once receiving a message from discord

**To Reproduce**
sending a message on discord

**Expected behavior**
make a bridge between discord and rocketchat
**Screenshots/debug logs**
```
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 77 [running]:
github.com/matterbridge/Rocket.Chat.Go.SDK/realtime.(*Client).SendMessage(0xc000006870, 0xc000540400, 0x19, 0x1b79d8e, 0x1)
        /home/travis/gopath/pkg/mod/github.com/matterbridge/!rocket.!chat.!go.!s!d!k@v0.0.0-20190210153444-cc9d05784d5d/realtime/messages.go:44 +0x1fd
github.com/42wim/matterbridge/bridge/rocketchat.(*Brocketchat).Send(0xc00030c3c0, 0xc000508fc0, 0x1a, 0xc000244f61, 0xa, 0xc0005089c0, 0x19, 0xc0002efca0, 0x12, 0xc000308ba0, ...)
        /home/travis/gopath/src/github.com/42wim/matterbridge/bridge/rocketchat/rocketchat.go:170 +0x68e
github.com/42wim/matterbridge/gateway.(*Gateway).SendMessage(0xc00030c060, 0xc000108410, 0xc000312180, 0xc0000fc380, 0x0, 0x0, 0xc, 0xc00003bf20, 0x17, 0xc00003bf20)
        /home/travis/gopath/src/github.com/42wim/matterbridge/gateway/gateway.go:458 +0x6b4
github.com/42wim/matterbridge/gateway.(*Gateway).handleMessage(0xc00030c060, 0xc000108410, 0xc000312180, 0x0, 0x0, 0x0)
        /home/travis/gopath/src/github.com/42wim/matterbridge/gateway/handlers.go:226 +0x24f
github.com/42wim/matterbridge/gateway.(*Router).handleReceive(0xc00030a000)
        /home/travis/gopath/src/github.com/42wim/matterbridge/gateway/router.go:149 +0x248
created by github.com/42wim/matterbridge/gateway.(*Router).Start
        /home/travis/gopath/src/github.com/42wim/matterbridge/gateway/router.go:102 +0x93e
```

**Environment (please complete the following information):**
 - OS: Windows10
 - Matterbridge version:version: 1.16.4-dev 4c44515

**Additional context**
```
[discord]
[discord.mydiscord]
Token="*"
Server="TEST"
RemoteNickFormat="[{PROTOCOL}] <{NICK}> "

[rocketchat.myrocketchat]
Server="*"
Login="111@test.com"
Password="test"
RemoteNickFormat="[{PROTOCOL}] <{NICK}> "

[[gateway]]
name="gateway1"
enable=true

[[gateway.inout]]
account="discord.mydiscord"
channel="test"

[[gateway.inout]]
account="rocketchat.myrocketchat"
channel="#chat_group"
```

**Comment by qaisjp** [comment]
Seems to be an issue here: https://github.com/matterbridge/Rocket.Chat.Go.SDK/blob/cf0caa70276455b003778afa1763194acccba604/realtime/messages.go#L33-L50

Not sure why the line numbers are a little off, might be just the debug info is stripped.

Upstream has some updates but it does not seem to be relevant: https://github.com/RocketChat/Rocket.Chat.Go.SDK/blob/f7adb9390c1a25a75ab86d2824ba5edc40ae51a4/realtime/messages.go#L63-L75

Labelling as `bug` and `library issue`. 

Basically looks like `rawResponse` is returning `nil` here:

```go
rawResponse, err := c.ddp.Call("sendMessage", m)
if err != nil {
	return nil, err
}

return getMessageFromData(rawResponse.(map[string]interface{})), nil
```

No idea how [DDP](https://blog.meteor.com/introducing-ddp-6b40c6aff27d) works but this is being called: https://github.com/gopackage/ddp/blob/652027933df47696cba320d7600af91a399ade18/ddp_client.go#L328-L332

**Comment by jv283** [comment]
This might be because of duplicate ID:s in new messages, fix should be quite simple: https://github.com/RocketChat/Rocket.Chat.Go.SDK/issues/40

**Comment by 42wim** [comment]
@jv283 if that's the case it shouldn't happen on the first message?

**Comment by kimmoal** [comment]
@42wim true, does not stop the fact that I am experiencing non-deterministic crashes with matterbridge with bridging IRC channels to RocketChat. I will make a separate issue, and I have already made a fix to upstream.

**Comment by 42wim** [comment]
@kimmoal if you want you can also make a PR against my fork  https://github.com/matterbridge/Rocket.Chat.Go.SDK/tree/matterbridge as I don't think upstream is very alive.

**Comment by 42wim** [comment]
Should be fixed in master, as in it won't crash on nil anymore.
